src/main/java/spoon/reflect/factory/TypeFactory.java

```diff
@@ -555,7 +555,7 @@ public CtTypeParameterReference createTypeParameterReference(String name) {
 	 * @param <T>
 	 * 		Type of the first bound.
 	 */
-	public <T> CtIntersectionTypeReference<T> createIntersectionTypeReferenceWithBounds(Set<CtTypeReference<?>> bounds) {
+	public <T> CtIntersectionTypeReference<T> createIntersectionTypeReferenceWithBounds(List<CtTypeReference<?>> bounds) {
```

Comment:
Can't you use an ordered `Set` based on the comparator used by the previous `TreeSet` ?